src/ansys/fluent/core/systemcoupling.py

```diff
@@ -0,0 +1,119 @@
+"""Module containing System Coupling-related functionality."""
+
+from dataclasses import dataclass
+import os
+from typing import List
+import xml.etree.ElementTree
+
+
+class SystemCoupling:
+    """Wrap a System Coupling object, adding methods to discover more about the
+    System Coupling related setup and to help solving a System Coupling analysis.
+
+    Methods
+    -------
+    __getattr__
+    get_variables
+    get_regions
+    get_analysis_type
+    syc_connect
+    syc_solve
+    """
+
+    @dataclass
+    class Variable:
+        name: str
+        display_name: str
+        tensor_type: str
+        is_extensive: bool
+        location: str
+        quantity_type: str
+
+    @dataclass
+    class Region(object):
+        name: str
+        display_name: str
+        topology: str
+        input_variables: List[str]
+        output_variables: List[str]
+
+    def __init__(self, solver):
+        self._solver = solver
+        self.participant_type = "FLUENT"
+
+    # def __getattr__(self, attr):
+    #    return getattr(self.solver, attr)
+
+    def get_variables(self):
```

Comment:
I would expect this function to be calling the API, not returning a dictionary that is constructed by writing the scp file.